app/flatpak-builtins-create-usb.c

```diff
@@ -484,6 +484,7 @@ flatpak_builtin_create_usb (int argc, char **argv, GCancellable *cancellable, GE
       g_auto(GStrv) parts = NULL;
       unsigned int j = 0;
       const char **arches;
+      FlatpakKinds first_ref_kind = 0;
```

Comment:
I think this and related variables would be clearer if you replaced `first` with `first_installed` and add a comment where `first_ref` is set to explain that we're just finding the first usable installed ref for the given partial ref (so it doesn't look like we're only checking the first ref passed on the command line for whether it's a runtime). (At least, assuming I'm understanding the code correctly).

Reading the context in greater detail, this is a little more clear without these changes but I honestly missed the outer loop upon first skim (starting where the changes occurred) because it's so long. In the future, I'd break up sections where feasible into separate static functions (even if they're only used once) so the top-level function (`flatpak_builtin_create_usb()`) is easier to understand conceptually. You could also add the `inline` keyword to deal with the minor performance hit of an extra function call but I'm pretty sure any modern compiler would default to doing that automatically for single-use functions. I also take the use of bare blocks in this function as an indicator that it's fairly long.

Reply:
I pushed a commit to rename `first_ref` and `first_ref_kind`. Hopefully it's clear enough now that it doesn't need a comment.

You're right, I probably could have broken up the main function a bit more.